### ISSUE TITLE
Code Review: SudokuFrame.java

### ISSUE THREAD

**Issue by 1to9only** [SudokuMonster/SukakuExplainer#85]
I've looked at some of the new code:

1. My view is that this code:
```
Settings.getInstance().setNC(mitRegularNC.isSelected() ? 1 : 0);
Settings.getInstance().toggleVariants();
Grid.changeVisibleCells();
if (mitRegularNC.isSelected())
    mitCyclicNC.setSelected(false);
```
makes it possible to have both RegularNC and CyclicNC set when calling toggleVariants(), I think it should look like this:
```
Settings.getInstance().setNC(mitRegularNC.isSelected() ? 1 : 0);
if (mitRegularNC.isSelected())
    mitCyclicNC.setSelected(false);
Settings.getInstance().toggleVariants();
Grid.changeVisibleCells();
```
I have not tested this.

2. It is also possible to have both Anti-King and and Anti-kNight set. I admit to not really knowng if an AK+AN sudoku exists.

3. In this code:
```
Settings.getInstance().toggleVariants();
Grid.changeVisibleCells();
```
changeVisibleCells() is redundant as it is called within toggleVariants().

4. In this code:
```
engine.clearGrid();
engine.clearHints();
```
clearHints() is redundant as it is called within clearGrid().

**Comment by SudokuMonster** [comment]
Thanks 1to9only,

1. On my desktop I think I've addressed this issue. I'll upload the latest code then check if it is still there. I will follow your suggestion if it hasn't changed and is still giving the same problem.
2. Yes that is possible and an AK AN is known variant
3. Thanks, noted
4. Thanks, noted.

I have on my desktop added new techniques that we discussed targeting NC. These should be up at some stage today

**Comment by SudokuMonster** [comment]
I've just checked my local code for the 1st issue. It looks like I've fixed it already!

This is under mitRegularNC and the opposite in under cyclicNC
```
if (mitRegularNC.isSelected())
	mitCyclicNC.setSelected(false);
Settings.getInstance().setNC(mitCyclicNC.isSelected() ? 2 : 0);					
Settings.getInstance().setNC(mitRegularNC.isSelected() ? 1 : 0);
Settings.getInstance().toggleVariants();
```

**Comment by SudokuMonster** [comment]
All points mentioned in this Issue have been addressed and would appear in all versions from v1.15.7 onwards
